src/mappings/mappingHandlers.ts

```diff
@@ -71,24 +73,21 @@ async function handlePool(pkeys: string[][], blockNumber: number, timestamp: Dat
   }
 }
 
-async function handleValue(vkeys: any[], blockNumber: number) {
+async function handleValue(valueKeys: [string, string][], blockNumber: number) {
   try {
     // wrap api query
-    let pars = [];
-    let valueQueries = [];
-    for (let k of vkeys) {
-      pars.push(k);
-      valueQueries.push(api.query.oracle.rawValues(...k));
-    }
-
-    const valueRes = await Promise.all(valueQueries);
+    const valueRes = await api.query.oracle.rawValues.multi(valueKeys);
+    const lpTokens = await getLpTokens();
+    const lpTokenAssetIds = lpTokens.map(i => i.id.toString());
 
     // group by assetId
     let groups = {};
-    for (let ind in pars) {
-      const assetId = pars[ind][1];
-      groups[assetId] = groups[assetId] || [];
-      groups[assetId].push(valueRes[ind]);
+    for (let index in valueKeys) {
+      const assetId = valueKeys[index][1];
+      if (![...lpTokenAssetIds, liquidStakingAssetId].includes(assetId)) {
```

Comment:
extract out the for-loop，no need to expand the arrays every time

Reply:
Do you mean to put [... .lpTokenAssetIds, liquidStakingAssetId] out of the for loop because of the performance of variable declarations in the loop?

Reply:
yes!